src/vehicles/RosVehicle.lua

```diff
@@ -0,0 +1,257 @@
+            scan_msg.angle_increment = spec.laser_scan_obj.LS_FOV / spec.laser_scan_obj.vehicle_table.laser_scan.num_rays
+            -- assume sensor gives 50 scans per second
+            scan_msg.time_increment = (1.0 / 50) / spec.laser_scan_obj.vehicle_table.laser_scan.num_rays
+            --scan_msg.scan_time = 0.0  -- we don't set this field (TODO: should we?)
+            scan_msg.range_min = spec.laser_scan_obj.vehicle_table.laser_scan.range_min
+            scan_msg.range_max = spec.laser_scan_obj.vehicle_table.laser_scan.range_max
+            scan_msg.ranges = spec.laser_scan_array
+            --scan_msg.intensities = {}  -- we don't set this field (TODO: should we?)
+
+            -- publish the message
+            pub_scan:publish(scan_msg)
+
+            -- convert to quaternion for ROS TF
+            -- note the order of the axes here (see earlier comment about FS chirality)
+            -- the rotation from base_link to raycastnode is the same as rotation from raycastnode to virtaul laser_frame_i as there is no rotation between base_link to raycastnode
+            local q = ros.Transformations.quaternion_from_euler(spec.laser_scan_obj.vehicle_table.laser_scan.laser_transform.rotation.z, spec.laser_scan_obj.vehicle_table.laser_scan.laser_transform.rotation.x, spec.laser_scan_obj.vehicle_table.laser_scan.laser_transform.rotation.y)
+
+            -- get the translation from base_link to laser_frame_i
+            -- laser_dy is the offset from laser_frame_i to laser_frame_i+1
+            local base_to_laser_x, base_to_laser_y, base_to_laser_z = localToLocal(spec.LaserFrameNode, self.components[1].node, 0, laser_dy, 0)
+
+            -- create single TransformStamped message
+            local tf_base_link_laser_frame_i = geometry_msgs_TransformStamped.new()
+            tf_base_link_laser_frame_i:set(
+                spec.base_link_frame,
+                t,
+                spec.base_link_frame .."/laser_frame_" .. i,
+                -- note the order of the axes here (see earlier comment about FS chirality)
+                base_to_laser_z,
+                base_to_laser_x,
+                base_to_laser_y,
+                -- we don't need to swap the order of q, since the calculation of q is based on the ROS chirality
+                q[1],
+                q[2],
+                q[3],
+                q[4]
+            )
+            self:addTF(tf_msg, tf_base_link_laser_frame_i)
+        end
+    end
+end
+
+
+function RosVehicle:getLaserFrameNode()
+    local spec = self.spec_rosVehicle
+
+    if self.spec_drivable then
+        local current_veh_name = ros.Names.sanatize(self:getFullName())
+        spec.laser_scan_obj = LaserScanner.new(self, mod_config.vehicle[current_veh_name])
+
+        local instance_veh = VehicleCamera:new(self, RosVehicle)
+        local xml_path = instance_veh.vehicle.configFileName
+        local xmlFile = loadXMLFile("vehicle", xml_path)
```

Comment:
This causes a memory leak (no delete), but you already know this, and it's fixed in a later commit (not part of this PR).